qoqo-qryd/src/tweezer_devices.rs

```diff
@@ -1478,6 +1534,60 @@ impl TweezerMutableDeviceWrapper {
             .set_default_layout(layout)
             .map_err(|err| PyValueError::new_err(format!("{:}", err)))
     }
+
+    /// Creates a graph representing a TweezerDevice.
+    ///
+    /// Args:
+    ///     draw_shifts (Optional[bool]): Whether to draw shifts or not. Default: false
+    ///     pixel_per_point (Optional[float]): The quality of the image.
+    ///     file_save_path (Optional[str]): Path to save the image to. Default: output the image with the display method.:
+    ///
+    /// Raises:
+    ///     PyValueError - if there is no layout, an error occurred during the compilation or and invalid path was provided.
+    ///
+    #[pyo3(text_signature = "(draw_shifts, pixel_per_point, file_save_path, /)")]
+    pub fn draw(
+        &self,
+        draw_shifts: Option<bool>,
+        pixel_per_point: Option<f32>,
+        file_save_path: Option<String>,
+    ) -> PyResult<()> {
+        let display_image = file_save_path.is_none();
+        let image = self
+            .internal
+            .draw(
+                pixel_per_point,
+                draw_shifts.unwrap_or(false),
+                file_save_path,
+            )
+            .map_err(|x| PyValueError::new_err(format!("Error during Circuit drawing: {x:?}")))?;
+
+        if display_image {
```

Comment:
I guess the variable instantiation above can be skipped, so `if file_save_path.is_none() { ...`